controllers/clusterca_to_secret.go

```diff
@@ -0,0 +1,190 @@
+package controllers
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	Namespace  = "global-trust-certificates"
+	SecretName = "cluster-ca-secret"
+)
+
+func ClusterCaToSecret(client client.Client) {
+	log := log.Log.WithName("ClusterCAToSecret")
+
+	ticker := time.NewTicker(20 * time.Second)
+	quit := make(chan struct{})
```

Comment:
The quit channel will never get a signal from here. You'll want to create the channel in main and pass it to this function. Then, main can send to the channel in that blocking mgr.Start(...) function just before the os.Exit call.